instrumentation/split-switches-pass.so.cc

```diff
@@ -452,7 +468,11 @@ bool SplitSwitchesTransform::splitSwitches(Module &M) {
     BranchInst::Create(SwitchBlock, OrigBlock);
 
     /* We are now done with the switch instruction, delete it. */
+#if LLVM_VERSION_MAJOR >= 16
+    // TODO to erase range of instructions instead ?
```

Comment:
I think that is the right thing to do.
to be sure you could check with a switch example for llvm < 16 and llvm >= 16 and see if the same instructions are generated?
I did not code this, so I do not know that well what happens here exactly and why ...